### ISSUE TITLE
Consider supporting multiple .NET framework versions

### ISSUE THREAD

**Issue by cnunciato** [pulumi/templates#269]
When we generate a new .NET project today with `pulumi new`, we target .NET 3.1:

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    ...
    <TargetFramework>netcoreapp3.1</TargetFramework>
    ...
</Project>
```

But with .NET 3.1 [a couple of versions behind now](https://dotnet.microsoft.com/en-us/download/dotnet) (6 is currently "recommended"), it's easy to run into problems if you're on a more recent version -- for example, I'm on 6, which was the default when I `brew install`ed a few days ago:

```
$ pulumi up
...

Diagnostics:
  pulumi:pulumi:Stack (registry-867-dev):
    It was not possible to find any compatible framework version
    The framework 'Microsoft.NETCore.App', version '3.1.0' (x64) was not found.
      - The following frameworks were found:
          6.0.3 at [/usr/local/Cellar/dotnet/6.0.103/libexec/shared/Microsoft.NETCore.App]
```

From https://docs.microsoft.com/en-us/nuget/create-packages/multiple-target-frameworks-project-file, it appears we may be able to support multiple versions of .NET:

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    ...
    <TargetFrameworks>netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
    ...
</Project>
```

But when I try this with `pulumi up`, while `dotnet build -nologo .` succeeds, the preview process fails with the following:

```
Diagnostics:
  pulumi:pulumi:Stack (registry-867-dev):
    Unable to run your project
    Your project targets multiple frameworks. Specify which framework to run using '--framework'.
```

Would be great to be able to support multiple versions if we can!

**Comment by Frassle** [comment]
Ah that's because we use `dotnet run` by default, and that won't know which of the frameworks to run with. I think the simpler thing here is we ought to just move up to dotnet 6.0. @Zaid-Ajaj did some work to unblock that, not sure if there's anything holding back just updating all the templates/examples now?

**Comment by Frassle** [comment]
N.B. We _could_ add a setting in the "runtime.options" of Pulumi.yaml to tell the dotnet plugin which framework version to pass to `dotnet run` but adding that to the templates just ends up hitting the same issue you ran into above. If we default to a version the user doesn't have, it will break.

I don't think we want to add .csproj and `dotnet --list-runtimes` parsing to try and work this out automatically.

**Comment by mikhailshilkov** [comment]
I can't reproduce it either - I have `6.0.300` sdk and our template `azure-csharp` builds fine for me.

Anyway, I suggest we switch our template to .NET 6 entirely and take advantage of the latest C# features to make it look as simple as possible (e.g. top-level statements). .NET 6 is the current LTS so it should be our default.

cc @Zaid-Ajaj

**Comment by mikhailshilkov** [comment]
Closing in favor of #292

**Comment by cnunciato** [comment]
This came up again recently [in an onboarding doc](https://docs.google.com/document/d/12ZHJZNyAJS-2RRV84I0cAdaRgcuVJQ6q8hKsHjax4bw/edit?disco=AAABPWjyb9c), so I went ahead and filed another issue here: https://github.com/pulumi/templates/issues/269
